docs/content/src/learn/tutorials/messaging-integrations/messaging-with-kafka/src/product_management_system/product_admin_service.bal

```diff
@@ -0,0 +1,53 @@
+import ballerina/http;
+import ballerina/kafka;
+import ballerina/io;
+
+// CODE-SEGMENT-BEGIN: kafka_producer_config
+kafka:ProducerConfig producerConfigs = {
+    bootstrapServers: "localhost:9092",
+    clientId: "kafka-producer",
+    acks: "all",
+    retryCount: 3
+};
+
+kafka:Producer kafkaProducer = new(producerConfigs);
+// CODE-SEGMENT-END: kafka_producer_config
+
+// HTTP Service Endpoint
+listener http:Listener httpListener = new(9090);
+
+@http:ServiceConfig { basePath: "/product" }
+service productAdminService on httpListener {
+
+    @http:ResourceConfig { methods: ["POST"], consumes: ["application/json"], produces: ["application/json"] }
+    resource function updatePrice(http:Caller caller, http:Request request) returns error? {
+        http:Response response = new;
+
+        json reqPayload = check request.getJsonPayload();
+        io:println("ProductManagementService : Received payload");
+
+        var productName = reqPayload.Product;
+        var productType = reqPayload.Type;
+        var productPrice = reqPayload.Price;
+
+        // Construct message to be published to the Kafka Topic
+        json productInfo = {
+            "Name" : productName.toString(),
+            "Price" : productPrice.toString()
+        };
+
+        // Serialize the message
+        byte[] serializedMessage = productInfo.toJsonString().toBytes();
```

Comment:
Rename `serializedMessage` to `kafkaMessage`

Reply:
Done